server/src/main/java/org/elasticsearch/indices/MapperMetrics.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices;
+
+import org.elasticsearch.common.util.ValueWithThreadLocalOverride;
+import org.elasticsearch.index.mapper.SourceFieldMetrics;
+
+/**
+ * Groups together all metrics used in mappers.
+ * Main purpose of this class is to avoid verbosity of passing individual metric instances around.
+ */
+public class MapperMetrics {
+    public static MapperMetrics NOOP = new MapperMetrics();
+
+    public static ValueWithThreadLocalOverride<SourceFieldMetrics> SOURCE_FIELD_METRICS = new ValueWithThreadLocalOverride<>(
```

Comment:
I think the use of `ValueWithThreadLocalOverride` isn't needed? I think `ThreadLocal.withInitial(...)` can be used here that reads from a static field. Then the test just changes the NOOP static field to point to another instance. Additionally the init method can be changed to update NOOP instance?

Reply:
If a test changes static field it potentially impacts other concurrent test using this static field. That is the main idea behind `ValueWithThreadLocalOverride`.

Reply:
`NOOP` here is indeed a leftover from previous implementation.